### DISCUSSION TITLE
V2.alpha to 2.x roadmap

### DISCUSSION THREAD

**Discussion by janklimo** [pickle-finance/pickle-ui#390] [General]
## 🚗  V2.alpha to 2.x roadmap

As discussed in the scrum call, this aims to be a very specific list of objectives that need to be met for specific UI release milestones.

### V2.alpha – community release (Apr 1st)

Intended to offer a core set of features to be tested by our loyal users to provide feedback on.

**pf-core**
- [x] Fix data issues in pf-core, issues [100](https://github.com/pickle-finance/pf-core/issues/100) and [104](https://github.com/pickle-finance/pf-core/issues/104).
- [x] Add pool IDs to API response
- [x] Fix user model, issue [154](https://github.com/pickle-finance/pf-core/issues/154)

**UI**
- [x] Dark and light theme
- [x] Fix any missing/ugly icons
- [x] Transaction flow animations from fennec
- [x] Token approval
- [x] Jar deposit
- [x] Jar withdrawal
- [x] pToken approval
- [x] Stake
- [x] Unstake
- [x] Harvest pickles (farm)
  - [x]  Apply the same flow to the harvest pickles modal on the dashboard
- [x] Lock dill
- [x] Extend dill lock
- [x] Harvest pickles (dill)

### V2 – public release (TBD)

With this release V2 becomes the default landing page with the option to visit V1. Feature-freeze of V1.

- [ ] CowSwap
- [x] Finalize dual distribution 
- [x] Make themes look good
  - [ ] Persist themes when reloading the page
- [x] Update routing to make V2 the default, add a link to go back to V1
- [x] Voting
- [x] Non-standard jars
  - [x] UniV3 jars
  - [x] B.Protocol
  - [x] Withdraw-only jars
  - [x] ~~Balancer~~
- [x] Get Mark's help on the final design touches. Voting, stats, farm details are too freestyle.

### V2.1 – follow-up release(s)

- [ ] Zaps
- [ ] Disabled/inactive jars (with filter)
- [ ] Depositing native assets (see #393) 
- [ ] Fix connectors on sidechains (using `@web3-react` v8)
- [x] Brinery, Brinery Pools, & Brinery Jars (https://github.com/pickle-finance/pickle-ui/issues/461)

**Comment by larrythecucumber321** [comment]
Thanks for the amazingly detailed checklist @janklimo 👏 

The items below are all for **V2 - public release**....

Amongst the list of Non-standard jars for the public release, I would include:
- [ ] Balancer timed-withdrawal jars
- [x] Withdraw-only jars
- [x] Disabled jars (which currently exist in the 'Inactive Jars' section of the UI)
- [ ] Depositing native assets, as done in #393 

Under Voting, we can further chunk those tasks into
- [x] Voting interaction
- [x] Vote distribution chart 

Not currently in production, but the FXS locking feature being developed in #191 wil need to be ported to V2.

That's all I can think of for now, but will add to this when/if other items come to me!

**Comment by manymoneymanny** [comment]
From Sprint planning, items to fix before release:
- [x] remove "compounding" indicator.
- [x] add `/dill/vote` page navigation
- [ ] add $PICKLE marketcap and total supply under price indicator or on hover
- [x] add APY breakdown on hover
- [x] add APY breakdown on details

**Reply by manymoneymanny** [reply]
> * [ ]  add `/dill/vote` page navigation

@bcstryker this one should be yours

**Reply by janklimo** [reply]
opened #469 and #470

**Comment by janklimo** [comment]
@mannyreimi88 or @bcstryker can you please add tickets for these?

**Reply by manymoneymanny** [reply]
testing DILL page today @janklimo:

some observations / suggestions:

- [x] add the $PICKLE balance in-wallet and the $PICKLEs locked in the current $DILL lock (and maybe their worth in $) so it's easy to grasp if I need to lock anything.
<img width="307" alt="how-many-pickles" src="https://user-images.githubusercontent.com/60312391/162998062-29278d3f-6ed4-46a3-9511-9eb4619b9fd4.png">

- [x] may need buttons to refresh balances in modals, this is because I am sure my balance changed while the page was already loaded but I had to refresh the whole page as when I clicked the modal by balance was zero.
<img width="449" alt="refresh-button" src="https://user-images.githubusercontent.com/60312391/162998546-5923ccd6-c185-4c42-afe7-2b6b32af6910.png">

- [x] `increase_amount()` call failed: https://etherscan.io/tx/0x75d57d3c1c9d45baa98f89bacf79b35b34635f15f8f1b3958ec1fb892795e92b
- [x] extend lock modal opens and default state seems there's a visual bug.
<img width="446" alt="slider-visual-bug-increase-lock" src="https://user-images.githubusercontent.com/60312391/162998820-80ad4d42-bc17-4847-9c79-bc41124383d4.png">

- [x] I was unsure whether I was extending lock for the max amount or not.
<img width="449" alt="max-lock-label" src="https://user-images.githubusercontent.com/60312391/162999376-8a3d6e61-b12c-4ee1-bc7c-90d2f7bbd2e2.png">

- [x] extend lock transaction couldn't be even relayed
<img width="405" alt="increase_unlock_time_total_fail" src="https://user-images.githubusercontent.com/60312391/163000295-96bc339e-0fbc-4237-9b2e-f33d0965c2c1.png">

- [x] gas estimates were off throughout
<img width="134" alt="gas-estimates-off" src="https://user-images.githubusercontent.com/60312391/162999523-94456863-fc1c-4ffa-bc44-1ae465f2ff62.png">

- [ ] more indicators are needed in the revenue distribution stats element
<img width="931" alt="revenue-stats-changes" src="https://user-images.githubusercontent.com/60312391/163000505-52154f6c-50be-496f-a969-a501afaf59e6.png">

**Reply by janklimo** [reply]
Thanks for testing this @mannyreimi88 very good points. @larrythecucumber321 I looked at the timestamp we pass to the increase lock transaction and it's 346 years into the future 😅 can you help fix some of these?

**Reply by manymoneymanny** [reply]
another cosmetic improvement would be if the "pie chart" image here we could use the "identicons" by Dan Finlay that are very classic to Ethereum https://www.npmjs.com/package/ethereum-identicon
<img width="171" alt="Screenshot 2022-04-13 at 1 10 44 AM" src="https://user-images.githubusercontent.com/60312391/163017091-49bf5920-edf8-4389-8a58-d23ccb4ed089.png">

**Reply by larrythecucumber321** [reply]
@janklimo I'll take care of these.

@mannyreimi88 Strange, I had no problems with these transactions testing myself just now... (see below). I have no idea how the increase lock transaction tried locking so far into the future, never seen that happen but I can investigate. Lastly, the failure to submit the transaction is a MetaMask issue. 

I can implement your other enhancements no problem. Can you transfer these onto an issue and a card on the project board please?
<img width="400" alt="image" src="https://user-images.githubusercontent.com/71284258/163488005-003e30ed-7842-4a24-9a12-6b50396ba9ca.png">

**Reply by manymoneymanny** [reply]
@larrythecucumber321 here you go https://github.com/pickle-finance/pickle-ui/issues/443
